src/ng/compile.js

```diff
@@ -943,6 +943,103 @@
  *
  * For information on how the compiler works, see the
  * {@link guide/compiler Angular HTML Compiler} section of the Developer Guide.
+ *
+ * @knownIssue
+ *
+ * ### Double Compilation
+ *
+ * Double compilation occurs when an already compiled part of the DOM gets compiled again. This is
+ * not an intended use case and can lead to misbehaving directives, performance issues, and memory
+ * leaks.
+ * A common scenario where this happens is a directive that calls `$compile` in a directive link
+ * function on the directive element:
+ *
+ * ```
+  angular.module('app').directive('addInput', function($compile) {
+    return {
+      link: function(scope, element, attrs) {
+        var newEl = angular.element('<input ng-model="$ctrl.value">');
+        attrs.$set('addInput', null) // To stop infinite compile loop
+        element.append(newEl);
+        $compile(element)(scope); // Double compilation
+      }
+    }
+  })
+  ```
+ * At first glance, it looks like removing the original `addInput` attribute is all there is needed
+ * to make this example work.
+ * However, if the directive element or its children have other directives attached, they will be compiled and
+ * linked again, because the compiler doesn't keep track of which directives have been assigned to which
+ * elements.
+ *
+ * This can cause unpredictable behavior, e.g. `ngModel` $formatters and $parsers will be
+ * attached again to the ngModelController. It can also degrade performance, as
```

Comment:
True, I was actually thinking of an issue with replace. I will change the example.